spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientAutoConfiguration.java

```diff
@@ -129,4 +144,102 @@ public InMemoryClientRegistrationRepository clientRegistrationRepository(
 			return new DefaultClientCredentialsTokenResponseClient();
 		}
 	}
+
+	private ClientHttpRequestInterceptor clientCredentialsTokenResolvingInterceptor(
+			ClientRegistration clientRegistration, ClientRegistrationRepository clientRegistrationRepository,
+			String clientId) {
+		Authentication principal = createAuthentication(clientId);
+		OAuth2AuthorizedClientService authorizedClientService = new InMemoryOAuth2AuthorizedClientService(
+				clientRegistrationRepository);
+		AuthorizedClientServiceOAuth2AuthorizedClientManager authorizedClientManager = new AuthorizedClientServiceOAuth2AuthorizedClientManager(
+				clientRegistrationRepository, authorizedClientService);
+		OAuth2AuthorizedClientProvider authorizedClientProvider = OAuth2AuthorizedClientProviderBuilder.builder()
+				.clientCredentials().build();
+		authorizedClientManager.setAuthorizedClientProvider(authorizedClientProvider);
+
+		OAuth2AuthorizeRequest authorizeRequest = OAuth2AuthorizeRequest
+				.withClientRegistrationId(DEFAULT_REGISTRATION_ID).principal(principal).build();
+
+		return (request, body, execution) -> {
+			OAuth2AuthorizedClient authorizedClient = authorizedClientManager.authorize(authorizeRequest);
+			request.getHeaders().setBearerAuth(authorizedClient.getAccessToken().getTokenValue());
+			return execution.execute(request, body);
+		};
+	}
+
+	private static final Authentication DEFAULT_PRINCIPAL = createAuthentication("dataflow-client-principal");
```

Comment:
Btw this and everything below (170-244) is a dead code for this PR. Code belongs to original azure work which needs additional code to get used.